### ISSUE TITLE
[BUG][KOTLIN] When generating apis and modals only using retrofit2 api files contain invalid import

### ISSUE THREAD

**Issue by ivanvasheka-intellias** [OpenAPITools/openapi-generator#15116]
##### Description

When generating code using kotlin generator with options to generate APIs and models files only, and using retrofit2 library, generator generates API files without infrastructure package, but API files contain imports of infrastructure classes which results into an error. In particular this one:

`import com.example.api.infrastructure.CollectionFormats.*`

##### openapi-generator version

6.3.0

##### OpenAPI declaration file content or url

task generateApi(type: GenerateTask) {
    generateApi.generatorName = "kotlin"
    generateApi.packageName = "com.example.api"
    generateApi.inputSpec = "api.json"
    generateApi.outputDir = "$rootDir/api"
    generateApi.skipValidateSpec = true

    generateApi.additionalProperties = [
            useCoroutines: "true",
            library      : "jvm-retrofit2",
            sourceFolder : "src/main/java",
    ]
    generateApi.globalProperties = [
            apiDocs  : "false",
            modelDocs: "false",
            apis     : "",
            models   : "",
    ]
}

##### Steps to reproduce

generate API files using provided options
open any API file
check imports

**Comment by kotovdv** [comment]
@ivanvasheka-intellias did you find a workaround for this? 

I am having the very same issue, but java + retrofit2 code generation options

**Comment by ivanvasheka-intellias** [comment]
> @ivanvasheka-intellias did you find a workaround for this?
> 
> I am having the very same issue, but java + retrofit2 code generation options

Yes. So the workaround for me was to allow generator to generate that exact file by adding this to the configuration:
```
task.globalProperties = [
    supportingFiles: "CollectionFormats.kt"
]
```
With this setting that file is generated and the import is valid despite the file not being used anywhere...

**Comment by parfentjev** [comment]
For those who, like me, came looking for an answer: these days, CollectionFormats requires StringUtil to compile. Fortunately, StringUtil does not add any extra dependencies.

The following configuration should do the trick:

```xml
<build>
    <plugins>
        <plugin>
            <groupId>org.openapitools</groupId>
            <artifactId>openapi-generator-maven-plugin</artifactId>
            <executions>
                <execution>
                    <id>api-codegen</id>
                    <goals>
                        <goal>generate</goal>
                    </goals>
                    <configuration>
                        <!-- other params -->
                        <supportingFilesToGenerate>
                            CollectionFormats.java,StringUtil.java
                        </supportingFilesToGenerate>
                    </configuration>
                </execution>
            </executions>
        </plugin>
    </plugins>
</build>
```

Hopefully this issue will be eventually addressed. If we configure a generator to skip support files, the skipped files should not be used as imports in clients and models. The import of CollectionFormats breaks retrofit interfaces, even though it is not used by them 🤷

**Comment by wing328** [comment]
@parfentjev would you like to contribute or sponsor a fix (to avoid unnecessary imports for use cases generating API and model files only)?
